utils/s3/client.cc

```diff
@@ -254,6 +254,7 @@ unsigned prepare_multipart_upload_parts(const utils::chunked_vector<sstring>& et
         if (etags.empty()) {
             return 0;
```

Comment:
unrelated, why etags.empty() lives inside the loop if etags is immutable? looks like we can move it outside.

Reply:
Oh, damn. It should've been `if (etag.empty())` instead to check for part upload failure :( Thanks!